sigmastate/src/main/scala/sigmastate/basics/BcDlogGroup.scala

```diff
@@ -1,28 +1,26 @@
 package sigmastate.basics
 
 import java.math.BigInteger
-import org.bouncycastle.asn1.x9.X9ECParameters
-import org.bouncycastle.crypto.ec.CustomNamedCurves
-import org.bouncycastle.math.ec.custom.sec.SecP256K1Point
-import org.bouncycastle.math.ec.ECPoint
-import org.bouncycastle.util.BigIntegers
+import sigmastate.crypto.BigIntegers
 import debox.cfor
+import sigmastate.crypto.{CryptoContext, CryptoFacade}
 
-import scala.collection.{Seq, mutable}
-import scala.util.Try
+import scala.collection.mutable
 
-abstract class BcDlogGroup[ElemType <: ECPoint](val x9params: X9ECParameters) extends DlogGroup[ElemType] {
 
-  lazy val curve = x9params.getCurve
+/** Base class for all dlog groups (with bouncycastle-like interface).
+  * @param ctx context which abstracts basic operations with curve and elements.
+  */
+abstract class BcDlogGroup(val ctx: CryptoContext) extends DlogGroup {
+  /** Modulus of the finite field of the underlying curve. */
+  lazy val p: BigInteger = ctx.getModulus
 
-  //modulus of the field
-  lazy val p: BigInteger = curve.getField.getCharacteristic
+  /** Order of the group (parameter N of the elliptic curve). */
```

Comment:
Notation is different from standard to standard, please provide a link to a standard "N" is taken from

Reply:
fixed